app/src/actioncommands.cpp

```diff
@@ -548,13 +548,29 @@ void ActionCommands::ZoomOut()
 void ActionCommands::rotateClockwise()
 {
     float currentRotation = mEditor->view()->rotation();
-    mEditor->view()->rotate(currentRotation + 15.f);
+    // Invert rotation direction if view is flipped either vertically or horizontally
+    if (mEditor->view()->isFlipHorizontal() ^ mEditor->view()->isFlipVertical())
+    {
+        mEditor->view()->rotate(currentRotation - 15.f);
```

Comment:
I think given that we have the same logic in two places now, we should simplify and refactor this to a new private function.

Something like private ActionCommands::rotateInDirection(int angle) { ... }

Reply:
I had a similar thought, but I think it would make more sense to have a rotate function in view that takes a relative angle, and maybe a bool for the direction (or just use the sign). The existing function could be changed or a new one could be created depending on if it is used anywhere else.

Reply:
Ah right, yeah that sounds fine to me too

Reply:
`ViewManager::rotate()` is used in:

![image](https://github.com/pencil2d/pencil/assets/75134774/83d5dcf8-cfbe-47de-b629-53c1e746620d)

Should I create something like `void rotateRelative(float delta);`?

Reply:
Here's what it looks like (please note that the shown functions can be improved too):

![image](https://github.com/pencil2d/pencil/assets/75134774/dfdddbe8-2de2-4ad1-b0b9-044c3b953632)

Reply:
Improved version

```C++
void ActionCommands::rotateClockwise()
{
    // Rotation direction is inverted if view is flipped either vertically or horizontally
    const float delta = mEditor->view()->isFlipHorizontal() == !mEditor->view()->isFlipVertical() ? -15.f : 15.f;
    mEditor->view()->rotateRelative(delta);
}

void ActionCommands::rotateCounterClockwise()
{
    // Rotation direction is inverted if view is flipped either vertically or horizontally
    const float delta = mEditor->view()->isFlipHorizontal() == !mEditor->view()->isFlipVertical() ? 15.f : -15.f;
    mEditor->view()->rotateRelative(delta);
}
```

Reply:
@MrStevns how does the `rotateRelative()` function above looks to you? can I commit that change?

it's essentially:
```C++
void ViewManager::rotateRelative(float delta)
{
    mRotation = mRotation + delta;
    updateViewTransforms();

    emit viewChanged();
}
```

Reply:
Sure, that's fine with me